hazelcast/src/main/java/com/hazelcast/map/impl/recordstore/AbstractRecordStore.java

```diff
@@ -208,4 +218,48 @@ public void disposeDeferredBlocks() {
     public Storage<Data, ? extends Record> getStorage() {
         return storage;
     }
+
+    @Override
+    public long getHits() {
+        return hits;
+    }
+
+    @Override
+    public long getLastAccessTime() {
+        return lastAccess;
+    }
+
+    @Override
+    public long getLastUpdateTime() {
+        return lastUpdate;
+    }
+
+    @Override
+    public void increaseHits() {
+        this.hits++;
+    }
+
+    @Override
+    public void setLastAccessTime(long time) {
+        this.lastAccess = Math.max(this.lastAccess, time);
+    }
+
+    @Override
+    public void setLastUpdateTime(long time) {
+        this.lastUpdate = Math.max(this.lastUpdate, time);
+    }
+
+
+    protected void updateStatsOnPut(boolean newRecord, long now) {
+        setLastUpdateTime(now);
+
+        if (!newRecord) {
+            updateStatsOnGet(now);
+        }
+    }
+
+    protected void updateStatsOnGet(long now) {
+        setLastAccessTime(now);
+        increaseHits();
```

Comment:
To me, if this PR is trying to preserve existing behaviour (it may be correct or not, discussion for  that is out of scope) and is only providing O(1) calculation for stats, `decrease` should be there because in current impl. after `map#clear`, there is no hit.